builder/builder/builder.go

```diff
@@ -395,7 +411,20 @@ func (b *Builder) Stop() error {
 	return nil
 }
 
-// BOLT: modify to calculate merkle inclusion proofs for preconfirmed transactions
+// GetConstraintsDomain returns the constraints domain used to sign constraints-API related messages.
+//
+// The builder signing domain is built as follows:
+// - We build a ForkData ssz container with the fork version and the genesis validators root. In the builder domain, this is an empty root.
```

Comment:
What do you mean with "In the builder domain, this is an empty root"?

Reply:
Across all implementations, the builder signing domain is built with an empty genesis validators root. I don't know why. In contrast, the beacon signing domain always contains a valid gvr.

Reply:
If it can be helpful, here is more context on the decision:

https://github.com/ethereum/builder-specs/issues/14#issuecomment-1127055996

> With that said, it makes the most sense to me to compute the domain for an application using `compute_domain(APP_DOMAIN_TYPE, fork_version=None, genesis_validators_root=None)`, because I don't think application should need any external data to compute / verify a signature.

Reply:
The builder domain does use the `fork_version` field though. I think for us that also makes sense to ensure signatures are only valid for the fork they're running on. Wdyt?